xcode/Subconscious/Shared/Library/Prose.swift

```diff
@@ -44,6 +44,37 @@ extension String {
         )
         return !truncated.isEmpty ? "\(truncated)\(ellipsis)" : ""
     }
+    
+    /// Take the first `maxBlocks` blocks or `Subtext.maxExcerptSize * maxBlocks` characters worth, terminating with an
+    /// ellipsis. If the string is truncated, it will be sliced to the nearest word boundary to avoid breaking markup.
+    /// This is used to implement `Subtext.excerpt`.
+    static func truncateAtWordBoundary(markup: any StringProtocol, maxChars: Int, fallback: String = "") -> String {
```

Comment:
This could be more efficiently defined as a generic instead of `any` arg.

No action necessary.